### DISCUSSION TITLE
Changing the license to MIT

### DISCUSSION THREAD

**Discussion by kayoub5** [chmorgan/sharppcap#487] [General]
Currently, sharppcap uses the LGPL license, this prevents using sharppcap in AOH and making self-contained apps, without making the whole application GPL or LGPL licensed.

This discussion is an attempt to get the approval from all the copyright holders and contributors of the project, to switch the license to MIT.

If your name is listed below, please write a comment to explicitly approve or decline the license change.

# Known Copyright Holders

| Who                                          | Approved  | Comment |
|----------------------------------------------|-|--|
| Ayoub Kaanich   <kayoub5@live.com>           | ✔️ | |
| Chris Morgan <chmorgan@gmail.com>            |  ✔️ | |
| Evan Plaice <evanplaice@gmail.com>           |  | Contribution limited to Tutorial and Examples in https://github.com/dotpcap/sharppcap/commit/b6e9899a9f8d528d8d700de9d336987f9c908e4a and https://github.com/dotpcap/sharppcap/commit/147bc908946068a10ec6803d769576ee97435b72 |
| Noah Potash   <noah.potash@outbreaklabs.com> | ✔️ | |
| Phillip Lemon   <lucidcomms@gmail.com>       | ✔️ | |
| Tamir Gal <tamir@tamirgal.com>               | ✔️| |

# Known Contributors

| Who               | Lines | Percent | Username| Approved  |
|-------------------|-------|---------|---|---|
| Chris Morgan      | 11012 | 52.36%  |  @chmorgan | ✔️  |
| Ayoub Kaanich     | 8611  | 40.94%  | @kayoub5  | ✔️ |
| Hendrik Eckardt   | 378   | 1.80%   | @heck-gd | ✔️  |
| Jan Pluskal       | 344   | 1.64%   | @pluskal  |  ✔️   |
| Lex Li            | 242   | 1.15%   | @lextm  |  ✔️ |
| houcem benali     | 114   | 0.54%   | @sensokame | ✔️  |
| Evan Plaice       | 109   | 0.52%   | @evanplaice |   |
| Noah Potash       | 51    | 0.24%   | @SapientGuardian  | ✔️  |
| Ilya Botov        | 38    | 0.18%   | @ARHARUP | ✔️  |
| Sterlinghv        | 37    | 0.18%   | @Sterlinghv | ✔️  |
| Cameron Elliott   | 34    | 0.16%   | @cameronelliott | ✔️  |
| paul              | 27    | 0.13%   |  @zpaul48  |   |
| Manuel Jacob      | 9     | 0.04%   | @MJay1337  |   |
| Viliam Letavay    | 8     | 0.04%   | @fukaminakrize | ✔️  |
| D10100111001      | 6     | 0.03%   | @D10100111001  |   |
| tamirgal          | 4     | 0.02%   | @tamirgal  | ✔️  |
| Mauro Levra       | 3     | 0.01%   |  @Nazardo | ✔️  |
| Trivalik          | 2     | 0.01%   | @trivalik  |  ✔️ |
| Fidel Perez-Smith | 1     | 0.00%   |  @fiddyschmitt | ✔️  |

# Verdict

Mozilla's lawyers believed it would be acceptable for Mozilla to make a good-faith effort to contract all contributors and then re-license, even if 5% of the contributors were unreachable (and none of the contributors disapproved).

you can read more about it in https://law.stackexchange.com/a/195

In our case, the contributors that voted, all voted in favor of changing the license, the ones that did not, have code contribution that consist of less than 1% of the current code base.

The copyright holders all agreed to change the license, except for "Evan Plaice" who is yet to reply, his contribution is limited to tutorial and documentation, thus does not consist a blocker for changing the library license.

Based on the votes, the small contribution size of the of the ones who did not vote, we may proceed on changing the project license, still I would give it a few days in hope that the remaining contributors do reply.

# Context

This discussion was originally started in #481

**Comment by fiddyschmitt** [comment]
Sounds great! Yes from me

**Comment by chmorgan** [comment]
Yes for me as well.

**Comment by sensokame** [comment]
Yes for me as well

**Comment by lextm** [comment]
Here I approve the license change for the portion of code I made.

**Comment by kayoub5** [comment]
@chmorgan do you happen to know what Tamir Gal username is?

**Comment by Sterlinghv** [comment]
Yes for me!

**Comment by cameronelliott** [comment]
I explicitly approve

**Comment by chmorgan** [comment]
I do, forwarded him the email and I’ll loop you in if he doesn’t reply to
the ticket and responds to me instead.

**Comment by ARHARUP** [comment]
Sure why not. I approve

**Comment by SapientGuardian** [comment]
I approve

**Comment by trivalik** [comment]
I approve

**Comment by kayoub5** [comment]
@chmorgan I see "Phillip Lemon" in the copyright notices, but no commits from him are recorded in git, any idea what did he contribute?

**Comment by chmorgan** [comment]
Hi Ayoub.

He provided permission to relicense his contributions to any license back
in 2009 when the lgpl license was made explicit. Can’t recall what the
contributions were. I have an email from him stating this. I can ask him if
he’s ok with me sharing the email with you, if you’d like something more
formal, but it’s been a while since we’ve had any communications,  it sure
if the email address is active etc.

Chris

**Comment by meck-gd** [comment]
I approve the license change.

**Comment by Nazardo** [comment]
Great idea, I approve.

**Comment by fukaminakrize** [comment]
I approve

**Comment by pluskal** [comment]
Definitely approve! Thanks :)

**Comment by tamirgal** [comment]
Hello, 
Great to see this project still kicking. I'm OK with the license change.
Thanks,
Tamir
